### PR TITLE
Command

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 1]
```diff
@@ -94,6 +90,46 @@ private void executeGrpcTask(GrpcTask task) {
...
-        task.execute(host, port);
+        GrpcTask grpcTask = new GrpcTask(host, Integer.valueOf(port));
+        grpcTask.execute(task);
+    }
+
+    public class GrpcTask extends AsyncTask<AbstractGrpcTaskExecutor, Void, String> {
+        private final String mHost;
+        private final int mPort;
+        private ManagedChannel mChannel;
+        private RoverServiceGrpc.RoverServiceBlockingStub stub;
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: Shouldn't this be mStub?

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 2]
```diff
@@ -94,6 +90,46 @@ private void executeGrpcTask(GrpcTask task) {
...
+
+        public GrpcTask(String mHost, int mPort) {
+            this.mHost = mHost;
+            this.mPort = mPort;
+        }
+
+        @Override
+        protected void onPreExecute() {
+            mChannel = ManagedChannelBuilder.forAddress(mHost, mPort)
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java:125]: +1 for onPreExecute

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 3]
```diff
@@ -56,35 +61,26 @@ public void onClick(View v) {
         mResultText = (TextView) findViewById(R.id.grpc_response_text);
     }
 
-    public void moveForward() {
+    private void moveForward() {
         ((InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE))
                 .hideSoftInputFromWindow(mHostEdit.getWindowToken(), 0);
-        GrpcTask movingRoverTask = new MovingRoverTask(this);
+        AbstractGrpcTaskExecutor movingRoverTask = new MovingRoverTask();
         executeGrpcTask(movingRoverTask);
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: nit: you can probably shorten this to `executeGrpcTask(new MovingRoverTask());`

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java (hunk 1 above)
Comment (ksheremet) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: Yes, thanks) (The same naming rules)

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 4]
```diff
@@ -94,6 +90,46 @@ private void executeGrpcTask(GrpcTask task) {
         mMoveForwardButton.setEnabled(false);
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: Since the buttons are re-enabled in postExecute, wouldn't it be fair to disable them in preExecute?

Alternatively, if you don't want them disabled asynchronously, how about you give the task a callback which would enable the buttons?

Ideally, it should be the same method e.g. `enableButtons(bool)` which goes through all the buttons, to avoid code duplicates and its consequences.

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java (hunk 1 above)
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: !fixed

**Comment by dotdoom** [comment]
Reviewed 1 of 1 files at r2.
Review status: 1 of 6 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dasfoo/rover-android-client/16)*

**Comment by dotdoom** [comment]
Reviewed 1 of 6 files at r1.
Review status: 2 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dasfoo/rover-android-client/16)*

**Comment by dotdoom** [comment]
Reviewed 1 of 6 files at r1.
Review status: 2 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dasfoo/rover-android-client/16)*

**Comment by dotdoom** [comment]
Reviewed 4 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dasfoo/rover-android-client/16)*

**Comment by dotdoom** [comment]
:shipit: LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/16/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Comment by dotdoom** [comment]
+1

**Comment by dotdoom** [comment]
:ship:

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/16/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)
